legate/core/_legion/util.py

```diff
@@ -416,3 +416,33 @@ def get_string(self) -> Optional[bytes]:
 
     def get_size(self) -> int:
         return self.size
+
+
+class Logger:
+    def __init__(self, name: str) -> None:
+        self.handle = legion.legion_logger_create(name.encode("utf-8"))
+
+    def __del__(self) -> None:
+        legion.legion_logger_destroy(self.handle)
+        self.handle = None
```

Comment:
There's no guarantee that a `__del__` will be called on interpreter exit, and if classes with`__del__` ever become part of an unreachable cycle, the presence of `__del__` makes the cycle uncollectible by the GC. I guess I'd ask:

* What are the consequences of not doing this at all?
* Could it be done in an `atexit` instead?

Reply:
> if classes with__del__ ever become part of an unreachable cycle, the presence of __del__ makes the cycle uncollectible by the GC

I think this no longer the case, as of python 3.4: https://docs.python.org/3/library/gc.html#gc.garbage

> There's no guarantee that a __del__ will be called on interpreter exit

Good point, I added an explicit "destructor", and calling it recursively during `Context` destruction.

> What are the consequences of not doing this at all?

Not very serious in this case, as we don't create very many of these Loggers. We would just end up leaking a small amount of state until process shutdown (logging messages should still get eagerly flushed).